### PR TITLE
fix: multitenancy and version related

### PR DESCRIPTION
**Pull request by sattvikc**
## Summary of change

(A few sentences about this PR)

## Related issues

- Link to issue1 here
- Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your
changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here
highlighting the necessary changes)

## Checklist for important updates

- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
    - In `build.gradle`
- [ ] If added a new paid feature, edit the `getPaidFeatureStats` function in FeatureFlag.java file
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them
  in `implementationDependencies.json`.
- [ ] Update function `getValidFields` in `io/supertokens/config/CoreConfig.java` if new aliases were added for any core config (similar to the `access_token_signing_key_update_interval` config alias).
- [ ] Issue this PR against the latest non released version branch.
    - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the
      latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/main/java/io/supertokens/webserver/api/multitenancy/CreateOrUpdateTenantOrGetTenantAPI.java [hunk 1]
```diff
@@ -50,7 +51,14 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
         String tenantId = InputParser.parseStringOrThrowError(input, "tenantId", true);
         if (tenantId != null) {
             tenantId = Utils.normalizeAndValidateTenantId(tenantId);
+
+            SemVer version = getVersionFromRequest(req);
+            if (version.lesserThan(SemVer.v3_0) && !tenantId.equals(TenantIdentifier.DEFAULT_TENANT_ID)) {
+                sendTextResponse(404, "Not found", resp);
+                return;
+            }
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/multitenancy/CreateOrUpdateTenantOrGetTenantAPI.java:59]:
```suggestion
           
```

File: src/main/java/io/supertokens/webserver/api/core/ApiVersionAPI.java [hunk 2]
```diff
@@ -45,11 +48,25 @@ protected boolean versionNeeded(HttpServletRequest req) {
     }
 
     @Override
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
         JsonObject result = new JsonObject();
         JsonArray versions = new JsonArray();
+
+        SemVer maxCDIVersion = getLatestCDIVersion();
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/webserver/api/core/ApiVersionAPI.java:55]: move the logic for the stuff below into the getLatestCDIVersion function

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/webserver/api/multitenancy/CreateOrUpdateTenantOrGetTenantAPI.java (hunk 1 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/multitenancy/CreateOrUpdateTenantOrGetTenantAPI.java:59]: Done

File: src/main/java/io/supertokens/webserver/api/core/ApiVersionAPI.java (hunk 2 above)
Comment (sattvikc) [reply src/main/java/io/supertokens/webserver/api/core/ApiVersionAPI.java:55]: Done

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**